calculator_simple/src/app_window.cpp

```diff
@@ -0,0 +1,307 @@
+                     , &btn_num_9_obj
+                     , &btn_num_a_obj
+                     , &btn_num_b_obj
+                     , &btn_num_c_obj
+                     , &btn_num_d_obj
+                     , &btn_num_e_obj
+                     , &btn_num_f_obj
+                  }}
+{
+   set_title("Simple Calculator");
+
+   btn_num_0_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_0), false);
+   btn_num_1_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_1), false);
+   btn_num_2_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_2), false);
+   btn_num_3_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_3), false);
+   btn_num_4_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_4), false);
+   btn_num_5_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_5), false);
+   btn_num_6_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_6), false);
+   btn_num_7_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_7), false);
+   btn_num_8_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_8), false);
+   btn_num_9_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_9), false);
+   btn_num_a_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_a), false);
+   btn_num_b_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_b), false);
+   btn_num_c_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_c), false);
+   btn_num_d_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_d), false);
+   btn_num_e_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_e), false);
+   btn_num_f_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_btn_f), false);
+
+   // btn_op_add_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_op_add));
+   // btn_op_sub_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_op_sub));
+   // btn_op_mul_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_op_mul));
+   // btn_op_div_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_op_div));
+   // btn_op_mod_obj.signal_clicked().connect(sigc::mem_fun(*this, &AppWindow::handle_op_mod));
+
+   radix_slider_obj.set_digits(0);
+   radix_slider_obj.set_has_origin(true);
+
+   radix_slider_obj.signal_change_value().connect(sigc::mem_fun(*this, &AppWindow::handle_radix_change), true);
+
+   layout_buttons.attach(radix_label_obj, 0, 0, 1, 1);
+   layout_buttons.attach(radix_slider_obj, 0, 1, 1, 4);
+
+   layout_buttons.attach(btn_num_0_obj, 1, 0, 1, 1);
+   layout_buttons.attach(btn_num_1_obj, 2, 0, 1, 1);
+   layout_buttons.attach(btn_num_2_obj, 3, 0, 1, 1);
+   layout_buttons.attach(btn_num_3_obj, 4, 0, 1, 1);
+   layout_buttons.attach(btn_op_add_obj, 5, 0, 2, 1);
```

Comment:
Note: the 2 isn't working since it's at the end of the row, so the width is collapsing and this button isn't any wider than the others.

Reply:
Resolved in 301f305.